EBI_indexing/update_index_json.py

```diff
@@ -17,35 +20,54 @@ def get_data_from_url(url, field=None):
-        "dataset7" : "https://blobserver.dc.scilifelab.se/blob/swedishpop_subplot_button.json/info.json",
-        "dataset8" : "https://blobserver.dc.scilifelab.se/blob/lineage_four_recent.json/info.json",
-        "dataset10" : "https://blobserver.dc.scilifelab.se/blob/wastewater_data_gu_allviruses.xlsx/info.json",
-        "dataset14" : "https://blobserver.dc.scilifelab.se/blob/SLU_wastewater_data.csv/info.json",
-        "dataset16" : "https://blobserver.dc.scilifelab.se/blob/wastewater_data_gu_allviruses.xlsx/info.json",
-        "dataset18": "https://blobserver.dc.scilifelab.se/blob/SLU_wastewater_data.csv/info.json"
+        "dataset1": "https://blobserver.dc.scilifelab.se/blob/Serology-testing-statistics-dataset-20202021.csv/info.json",
+        "dataset2": "https://blobserver.dc.scilifelab.se/blob/CSSS_estimates_mostrecent.csv/info.json",
+        "dataset3": "https://raw.githubusercontent.com/ScilifelabDataCentre/pathogens-portal/main/data/publications.json",
+        "dataset6": "https://blobserver.dc.scilifelab.se/blob/accompdiag_table_swe.json/info.json",
+        "dataset7": "https://blobserver.dc.scilifelab.se/blob/swedishpop_subplot_button.json/info.json",
+        "dataset8": "https://blobserver.dc.scilifelab.se/blob/lineage_four_recent.json/info.json",
+        "dataset10": "https://blobserver.dc.scilifelab.se/blob/wastewater_data_gu_allviruses.xlsx/info.json",
+        "dataset12": "https://blobserver.dc.scilifelab.se/blob/stockholm_wastewater_method_Sep_2021.xlsx/info.json",
+        "dataset14": "https://blobserver.dc.scilifelab.se/blob/SLU_wastewater_data.csv/info.json",
+        "dataset16": "https://blobserver.dc.scilifelab.se/blob/wastewater_data_gu_allviruses.xlsx/info.json",
+        "dataset18": "https://blobserver.dc.scilifelab.se/blob/SLU_wastewater_data.csv/info.json",
+        "dataset19": ["https://blobserver.dc.scilifelab.se/blob/KTH-produced-antigens.xlsx/info.json",
+                      "https://blobserver.dc.scilifelab.se/blob/External-PLP-proteinlist.xlsx/info.json"]
     }
 
     # Iterate thorugh the above dict and get recent modifed dates. For all files
-    # in blobserver, the info.json should have modified field in speciifc format
+    # in blobserver, the info.json should have modified field in specific format
     for key, url in info_urls.items():
-        info_to_update[key + "_modified"] = get_data_from_url(url, field="modified")[2:10]
-    new_index_data = json_templete.format(**info_to_update)
+        # for dataset3 (publication data), the key is different
+        field_to_get = "timestamp" if key == "dataset3" else "modified"
+        # for dataset19 (multi-disease serology), we have two file to check the data
+        if key == "dataset19":
+            u_dates = []
+            for u in url:
+                u_dates.append(datetime.strptime(get_data_from_url(u, field=field_to_get)[2:10], "%y-%m-%d"))
+                info_to_update[key + "_modified"] = max(u_dates).strftime("%y-%m-%d")
```

Comment:
Here, it should be one level up, so it find the `max` after the `for`. 

Maybe I should refresh my `PEP8` about new lines.

Reply:
I guess it's my auto-indent that went crazy here, I still have to setup some things in my dev environment.
Will fix it